store/copr/coprocessor.go

```diff
@@ -180,7 +181,8 @@ func buildCopTasks(bo *Backoffer, cache *RegionCache, ranges *KeyRanges, req *kv
 
 	rangesLen := ranges.Len()
 
-	locs, err := cache.SplitKeyRangesByLocations(bo, ranges)
+	// TODO(youjiali1995): is there any request type that needn't be splitted by buckets?
```

Comment:
Will the split affect some internal work such as br cheksum, data analyze or statistic update?

Reply:
Yes, they should be splitted by buckets too.

Reply:
Yes. TiFlash has implemented server-side streaming by `BatchCoprocessor` which splits ranges by segments.